app/src/main/java/com/example/mpvcalculatorrx/mvp/model/CalculatorModel.java

```diff
@@ -0,0 +1,109 @@
+package com.example.mpvcalculatorrx.mvp.model;
+
+import java.util.ArrayList;
+
+public class CalculatorModel extends MvpModel {
+    private ArrayList<Object> buffer;
+    private boolean complement = false;
+    private long result;
+
+    public CalculatorModel() {
+        buffer = new ArrayList<>();
+    }
+
+    public void addDigit(final long digit) {
+        int lastBufferIndex = buffer.size() - 1;
+        long value;
+        if (lastBufferIndex > -1) {
+            Object lastObject = buffer.get(lastBufferIndex);
+
+            if (lastObject instanceof Long) {
+                value = (Long) lastObject;
+
+                boolean c = value < 0;
+                value = c ? value * -1 : value;
+                value = value * 10 + digit;
+                value = c ? value * -1 : value;
+
+                buffer.set(lastBufferIndex, value);
+            } else {
+                if (buffer.size() < 3) {
+                    value = digit;
+                    if (complement) {
+                        value = digit * -1;
+                    }
+                    complement = false;
+                    buffer.add(value);
+                }
+            }
+        } else {
+            value = digit;
+            if (complement) {
+                value = digit * -1;
+            }
+            complement = false;
+            buffer.add(value);
+        }
+    }
+
+    public void addOperator(final Character operator) {
+        if (buffer.size() < 3) {
+            int lastBufferIndex = buffer.size() - 1;
+
+            if (lastBufferIndex > -1) {
+                Object lastItem = buffer.get(lastBufferIndex);
+                if (lastItem instanceof Long) {
+                    buffer.add(operator);
+                } else if (operator == '-') {
+                    complement = true;
+                }
+            } else if (operator == '-') {
+                complement = true;
+            }
+        }
+    }
+
+    public void reset() {
+        buffer = new ArrayList<>();
+        result = 0;
+    }
+
+    public void calculate() {
+        if (buffer.size() != 3) {
+            return;
+        }
+        Long result = 0L;
```

Comment:
It's better if you rename this variable to avoid use "this".

Reply:
Done